src/com/teamcautionrobotics/robot2018/Robot.java

```diff
@@ -87,6 +93,31 @@ public void teleopPeriodic() {
             spinSpeed = manipulator.getAxis(Axis.RIGHT_TRIGGER);
         }
         intake.spin(spinSpeed);
+
+        boolean towerRaiseStateButton = manipulator.getButton(Button.Y);
+        if (towerRaiseStateButton != towerRaiseStatePressed) {
+            if (towerRaiseStateButton) {
+                towerRaisingState = !towerRaisingState;
+            }
+            towerRaiseStatePressed = towerRaiseStateButton;
+        }
+
+        if (towerRaisingState) {
+            tower.setPosition(tower.getCurrentTowerPosition().next());
+            towerRaisingState = false;
+        }
+
+        boolean towerLowerStateButton = manipulator.getButton(Button.Y);
```

Comment:
Everything I said of the code to raise the lift applies to this code to lower the lift.